src/vsphere_cpi/lib/cloud/vsphere/vm_config.rb

```diff
@@ -11,8 +19,28 @@ def upgrade_hw_version?(vmtype_hw_version, global_hw_version)
       vmtype_hw_version.nil? ? global_hw_version : vmtype_hw_version
     end
 
+    # NOTE : This always creates a new name
+    def get_uuid_name
+      @vm_cid = "vm-#{SecureRandom.uuid}"
+      @vm_cid
+    end
+
+    # This returns if the name is already cached.
     def name
-      @vm_cid ||= "vm-#{SecureRandom.uuid}"
+      return @vm_cid if @vm_cid
+      if human_readable_name_enabled? && human_readable_name_info.has_key?('instance_group_name') && human_readable_name_info.has_key?('deployment_name')
+        @vm_cid = generate_human_readable_name(human_readable_name_info['instance_group_name'], human_readable_name_info['deployment_name'])
+      elsif human_readable_name_enabled?
+          logger.info("Not enough metadata to construct human readable name, using UUID based naming.")
+          @vm_cid = get_uuid_name
```

Comment:
Putting the default case in `else` should be avoided. Generally methods should return with defaults early.